pkg/cli/cli_test.go

```diff
@@ -312,6 +312,44 @@ func TestDeleteHostsCommandFails(t *testing.T) {
 
 }
 
+func TestCreateCVDLocalImageMissingEnvVariables(t *testing.T) {
```

Comment:
This test seems wrong to me:
1. It depends on an implementation detail: the fact that the env variables are checked before everything else instead of when needed.
2. If the code is modified to depend on a new variable this test won't be affected and there is no indication to the developer that they should add that new variable here.
3. It's not exactly incorrect to not read all 3 variables, but if you decide that 2 are enough then the test will fail even when the code is correct.

A more useful test would be to ensure the function doesn't fail when the 3 variables are set. You could create a temporary directory structure with the expected files and mock the `apiClient` to do nothing on upload to make a test like that work.

Reply:
agree, removed the test, added new exported method as suggested down below and added unit test to them